app.rb

```diff
@@ -2,34 +2,41 @@
 require_relative './lib/student'
 require_relative './lib/teacher'
 require_relative './lib/rental'
+require_relative 'storage'
 
 class Option
   def initialize
-    @books = []
-    @person = []
-    @rents = []
+    @reserve_people = Storage.new('data/people.json')
+    @reserve_books = Storage.new('data/books.json')
+    @reserve_rentals = Storage.new('data/rentals.json')
+    @books = load_books_data || []
+    @people = load_person_data || []
+    @rentals = @reserve_rentals.load_data || []
   end
 
-  def list_books
-    puts "There is no book yet! Please create a new book first. \n" if @books.empty?
-
-    @books.each do |book|
-      puts "Title: #{book.title}, Author: #{book.author}"
+  def list_all_books
+    if @books.empty?
+      puts 'No books found!'
+    else
+      @books.each do |book|
+        puts "Title: #{book.title}   Author: #{book.author}"
+      end
     end
   end
 
-  def list_people
-    puts "There is no person yet! Please create a new person first \n" if @person.empty?
-    @person.each do |each|
-      puts " [#{each.class}] Name: #{each.name}, ID: #{each.id}, Age: #{each.age}"
+  def list_all_people
+    if @people.empty?
+      puts 'No person found!'
+    else
+      @people.each do |person|
```

Comment:
Same here, when I try to list all books from storage I get an error. This is because you renamed the method `list_people` to `list_all_people`. Kindly make sure that you are calling the right method in order to list people successfully.

![image](https://github.com/GraceHtet/school_library/assets/38088818/d186096f-5a2d-45b9-8c39-2dc925f75fb1)

Reply:
well resolved! ✔️